scripts/genllheaders/genllheaders.py

```diff
@@ -0,0 +1,101 @@
+"""
+Utility to autogenerate generic LL HAL headers.
+
+Usage::
+
+    python3 genllheaders.py [-p /path/to/HAL] [-o /path/to/output_dir]
+
+Copyright (c) 2020 Teslabs Engineering S.L.
+
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import argparse
+from pathlib import Path
+import re
+import shutil
+
+from jinja2 import Environment, FileSystemLoader
+
+
+SCRIPT_DIR = Path(__file__).absolute().parent
+"""Script directory."""
+
+TEMPLATE_FILE = "header-template.j2"
+"""Template file."""
+
+REPO_ROOT = SCRIPT_DIR / ".." / ".."
+"""Repository root (used for input/output default folders)."""
+
+LL_API_IGNORE = [
+    "usb"
+]
+"""List of LL APIs to be ignored."""
+
+
+def main(hal_path, output):
+    """Entry point.
+
+    Args:
+        hal_path: Zephyr CubeMX HAL path.
+        output: Output directory.
+    """
+
+    if output.exists():
+        shutil.rmtree(output)
+    output.mkdir(parents=True)
+
+    # obtain all available LL APIs for each family
+    ll_apis = dict()
+    for entry in sorted(hal_path.iterdir()):
+        if not entry.is_dir() or not entry.name.startswith("stm32"):
+            continue
+
+        family = entry.name
+
+        family_headers = entry / "drivers" / "include"
+        for header in family_headers.iterdir():
+            m = re.match(r"stm32[a-z0-9]+_ll_([a-z0-9]+)\.h", header.name)
+            if not m:
+                continue
+
+            ll_api = m.group(1)
+
+            if ll_api in LL_API_IGNORE:
+                continue
+
+            if ll_api not in ll_apis:
+                ll_apis[ll_api] = list()
+            ll_apis[ll_api].append(family)
+
+    # write header files
+    env = Environment(
+        trim_blocks=True, lstrip_blocks=True, loader=FileSystemLoader(SCRIPT_DIR)
+    )
+    template = env.get_template(TEMPLATE_FILE)
+
+    for ll_api, families in ll_apis.items():
+        header_file = output / ("stm32_ll_" + ll_api + ".h")
+        with open(header_file, "w") as f:
+            f.write(template.render(ll_api=ll_api, families=families))
+
```

Comment:
Maybe new line write missing here? Thus you can then avoid styl checker warnings (see my commits below).

Reply:
No new line is needed here, added to Jinja template.